hw_2.js

```diff
@@ -0,0 +1,231 @@
+    delete user.first_name;
+    delete user.last_name;
+    if (!result[userGender]) {
+      result[userGender] = [user];
+    } else {
+      result[userGender].push(user);
+    }
+    return result;
+  }, {});
+};
+console.log(sortByGender(users));
+
+// 2.Преобразуйте массив в объект используя функцию reduce.
+
+const videos = [
+  {
+    id: 65432445,
+    title: "The Chamber",
+  },
+  {
+    id: 675465,
+    title: "Fracture",
+  },
+  {
+    id: 70111470,
+    title: "Die Hard",
+  },
+  {
+    id: 654356453,
+    title: "Bad Boys",
+  },
+];
+
+// Ожидаемый результат...
+//     {
+//         "65432445": "The Chamber",
+//         "675465": "Fracture",
+//         "70111470": "Die Hard",
+//         "654356453": "Bad Boys"
+//     }
+
+const createObjectFromArray = (videos) => {
+  return videos.reduce((result, video) => {
+    result[video.id] = video.title;
+    return result;
+  }, {});
+};
+console.log(createObjectFromArray(videos));
+
+//  3. Выведите массив ids для релизов у которых рейтинг 5.0.
+
+const newReleases = [
+  {
+    id: 70111470,
+    title: "Die Hard",
+    boxart: "http://cdn-0.nflximg.com/images/2891/DieHard.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [4.0],
+    bookmark: [],
+  },
+  {
+    id: 654356453,
+    title: "Bad Boys",
+    boxart: "http://cdn-0.nflximg.com/images/2891/BadBoys.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [5.0],
+    bookmark: [{ id: 432534, time: 65876586 }],
+  },
+  {
+    id: 65432445,
+    title: "The Chamber",
+    boxart: "http://cdn-0.nflximg.com/images/2891/TheChamber.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [4.0],
+    bookmark: [],
+  },
+  {
+    id: 675465,
+    title: "Fracture",
+    boxart: "http://cdn-0.nflximg.com/images/2891/Fracture.jpg",
+    uri: "http://api.netflix.com/catalog/titles/movies/70111470",
+    rating: [5.0],
+    bookmark: [{ id: 432534, time: 65876586 }],
+  },
+];
+
+// // [675465, …]
+
+const getIds = (newReleases) => {
+  return newReleases.reduce((result, newRelease) => {
+    if (newRelease.rating[0] === 5.0) {
+      result.push(newRelease.id);
+    }
+    return result;
+  }, []);
+};
+console.log(getIds(newReleases));
+
+// 4. С помощью функций map, reduce, вывести url у которого самая большая площадь
+
+const boxarts = [
+  {
+    width: 200,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture200.jpg",
+  },
+  {
+    width: 150,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture150.jpg",
+  },
+  {
+    width: 300,
+    height: 200,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture300.jpg",
+  },
+  {
+    width: 425,
+    height: 150,
+    url: "http://cdn-0.nflximg.com/images/2891/Fracture425.jpg",
+  },
+];
+
+const getMaxAreaUrl = (boxarts) => {
+  const maxArea = Math.max(
+    ...boxarts.map((boxart) => boxart.width * boxart.height)
+  );
```

Comment:
redundant code. you can do the square calculatuion inside reduce on each iteration. This will benefit you in case to not iterate through the array twice